desc/objectives/linear_objectives.py

```diff
@@ -39,6 +40,223 @@ def update_target(self, eq):
             self.jit()
 
 
+# TODO: make this work with above, but for multiple target args?
+class FixParameter(_Objective):
+    """Fix specific degrees of freedom associated with a given Optimizable object.
+
+    Parameters
+    ----------
+    things : Optimizable
+        Object whose degrees of freedom are being fixed.
+    params : str or list of str
+        Names of parameters to fix. Defaults to all parameters.
+    target : dict of {float, ndarray}, optional
+        Target value(s) of the objective. Only used if bounds is None.
+        Should have the same tree structure as thing.params. Defaults to things.params.
+    bounds : tuple of dict {float, ndarray}, optional
+        Lower and upper bounds on the objective. Overrides target.
+        Should have the same tree structure as thing.params.
+    weight : dict of {float, ndarray}, optional
+        Weighting to apply to the Objective, relative to other Objectives.
+        Should be a scalar or have the same tree structure as thing.params.
+    normalize : bool, optional
+        Whether to compute the error in physical units or non-dimensionalize.
+        Has no effect for this objective.
+    normalize_target : bool, optional
+        Whether target and bounds should be normalized before comparing to computed
+        values. If `normalize` is `True` and the target is in physical units,
+        this should also be set to True. Has no effect for this objective.
+    name : str, optional
+        Name of the objective function.
+
+    """
+
+    _scalar = False
+    _linear = True
+    _fixes = True
+    _units = "(~)"
+    _print_value_fmt = "Fixed parameter error: {:10.3e} "
+
+    def __init__(
+        self,
+        thing=None,
+        params=None,
+        target=None,
+        bounds=None,
+        weight=1,
+        normalize=False,
+        normalize_target=False,
+        name="Fixed parameter",
+    ):
+
+        self._target_from_user = target
+        self._params = params
+        super().__init__(
+            things=thing,
+            target=target,
+            bounds=bounds,
+            weight=weight,
+            normalize=normalize,
+            normalize_target=normalize_target,
+            name=name,
+        )
+
+    def build(self, thing=None, use_jit=False, verbose=1):
+        """Build constant arrays.
+
+        Parameters
+        ----------
+        use_jit : bool, optional
+            Whether to just-in-time compile the objective and derivatives.
+        verbose : int, optional
+            Level of output.
+
+        """
+        self.things = setdefault(thing, self.things)
+        thing = self.things[0]
```

Comment:
why is it always the first element of things?

Reply:
This objective should only ever be used with a single optimizable object